python/mxnet/symbol/symbol.py

```diff
@@ -1437,6 +1437,37 @@ def _gen_atomic_symbol(self):
         return Symbol(handle)
 
 
+    def optimize_for(self, backend, ctx=None, args=None, **kwargs):
+        """Partition symbol and optimize it for a given backend"""
+        out = SymbolHandle()
+        if ctx is None or args is None:
+            infer_shape_type = False
+        else:
+            infer_shape_type = True
+            if not isinstance(ctx, Context):
+                raise TypeError("Context type error")
+            listed_arguments = self.list_arguments()
+            args_handle, args = self._get_ndarray_inputs('args', args, listed_arguments, False)
```

Comment:
are we overwriting "args" here thats the input to the optimize_for function?

Reply:
It's suppose to be that way. Similar to: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/symbol/symbol.py#L1799